docs/tutorial/flask.rst

```diff
@@ -0,0 +1,530 @@
+    `Discourse <https://discourse.charmhub.io/>`_ (or follow the
+    "Edit this page on GitHub" on the bottom of
+    this document to comment directly on the document).
+
+
+Set things up
+=============
+
+.. include:: /reuse/tutorial/setup_stable.rst
+
+Finally, create a new directory for this tutorial and go inside it:
+
+.. code-block:: bash
+
+    mkdir flask-hello-world
+    cd flask-hello-world
+
+Create the Flask application
+============================
+
+Start by creating the "Hello, world" Flask application that will be
+used for this tutorial.
+
+Create a ``requirements.txt`` file, copy the following text into it
+and then save it:
+
+.. literalinclude:: code/flask/requirements.txt
+
+In the same directory, copy and save the following into a text file
+called ``app.py``:
+
+.. literalinclude:: code/flask/app.py
+    :language: python
+
+Run the Flask application locally
+=================================
+
+Install ``python3-venv`` and create a virtual environment:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-venv]
+    :end-before: [docs:create-venv-end]
+    :dedent: 2
+
+Now that we have a virtual environment with all the dependencies, let's
+run the Flask application to verify that it works:
+
+.. code-block:: bash
+
+    flask run -p 8000
+
+Test the Flask application by using ``curl`` to send a request to the root
+endpoint. You may need a new terminal for this; if you are using Multipass
+use ``multipass shell charm-dev`` to get another terminal:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:curl-flask]
+    :end-before: [docs:curl-flask-end]
+    :dedent: 2
+
+The Flask application should respond with ``Hello, world!``. The Flask
+application looks good, so we can stop for now using
+:kbd:`Ctrl` + :kbd:`C`.
+
+Pack the Flask application into a rock
+======================================
+
+First, we'll need a ``rockcraft.yaml`` file. Rockcraft will automate its
+creation and tailoring for a Flask application by using the
+``flask-framework`` profile:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+The ``rockcraft.yaml`` file will automatically be created and set the name
+based on your working directory. Open the file in a text editor and check
+that the ``name`` is ``flask-hello-world``. Ensure that ``platforms``
+includes the architecture of your host. For example, if your host uses the
+ARM architecture, include ``arm64`` in ``platforms``.
+
+.. note::
+
+    For this tutorial, we'll use the ``name`` "flask-hello-world" and assume
+    you are on the ``amd64`` platform. Check the architecture of your system
+    using ``dpkg --print-architecture``. Choosing a different name or
+    running on a different platform will influence the names of the files
+    generated by Rockcraft.
```

Comment:
I think this is immediately useful information that need not be dressed as a note. I might even put it in the introduction, too, to help set the stage for what we'll accomplish.

Reply:
I am fine with removing this note, but I do think the information about the system architecture is useful.

For the Django version of this tutorial, Daniele had suggested turning this note into a step, i.e., have all users run ``dpkg --print-architecture`` and compare it to the ``platforms`` section in the YAML file. 

The change would remove this note, and the paragraph above would be altered slightly. Let me know if you have any thoughts or concerns about this plan.

Reply:
I'm not suggesting we remove it, only that we promote it to a normal paragraph.

I don't see a way of converting the "we'll use the name" bit into a step, but yes, the rest could be steps. I'll reach out to you offline about this as well.

Reply:
I guess that I was suggesting removing the note because the previous paragraph reiterates a lot of the content in the note (including the `name`), and I agree with Daniele's suggestion to have the users run `dpkg --print-architecture` and compare it to `platforms` in the YAML file. I believe that the updated commands (taken from the spread test) mean that we no longer assume the `amd64` platform.

Reply:
Works for me!